### PR TITLE
Remove unexpected p.sync-line in code blocks nested into fenced code blocks

### LINKED ISSUE #37: Unexpected p.sync-line in code blocks nested into fenced code blocks
When a fenced code block contains another code block example, the preview unexpectedly renders `<p data-line="?" class="sync-line" style="margin:0;"></p>`. This happens no matter what language is present after backticks (if any). Indented code blocks are not affected.

The bug is observed both in Atom and VSCode.

screen shot 2018-02-12 at 16 37 22

## Copy-pastable minimal working example (MWE)

`````markdown
### Fenced markdown source

````markdown
# Header in source

```javascript
console.log('nested fenced block')
```

    console.log('nested indented block')

more text
````

### Fenced plain source

````
# Header in source

```javascript
console.log('nested fenced block')
```

    console.log('nested indented block')

more text
````

### Indented source

    # Header in source

    ```javascript
    console.log('nested fenced block')
    ```

        console.log('nested indented block')

    more text

End of MWE
`````

### PR DESCRIPTION
**Pull request by kachkaev**
Closes #37.

I also remove one useless declaration from tsconfig given the opportunity 🙂

### CODE REVIEW

**Comment by shd101wyy** [comment]
@kachkaev Oh sorry I post a comment sometime ago and I just realized that it was pending...

screen shot 2018-05-22 at 11 45 42 am

**Comment by shd101wyy** [comment]
@kachkaev Feel free to merge it. Sorry for the wait.

File: src/transformer.ts [hunk 1]
```diff
@@ -288,18 +288,27 @@ export async function transformMarkdown(
...
+        if (!inCodeBlock && !notSourceFile && containsCmd) {
           // it's code chunk, so mark its offset
           line = line.replace("{", `{code_chunk_offset=${codeChunkOffset}, `);
           codeChunkOffset++;
         }
-        inBlock = !inBlock;
+        if (!inCodeBlock) {
+          lastOpeningCodeBlockFence = currentCodeBlockFence;
+        } else if (
+          currentCodeBlockFence.length >= lastOpeningCodeBlockFence.length
```
[the remark this answers is not in the document]
Comment (kachkaev) [reply src/transformer.ts]: To close a code block, the fence we are currently looking at should be longer than the opening one or of the same length. For instance,

``````markdown
````js
````                          <- this one would close the ````markdown block
``````

``````markdown
````markdown
`````                         <- this one would also close the ````markdown block
``````

``````markdown
````markdown
```                           <- this one would NOT close the ````markdown block
``````

use case:

``````markdown
````markdown
some markdown here

```js                         <-- should not close ````markdown
// some js here
```                           <-- should not close ````markdown

````                          <-- should close ````markdown
``````

Does this look right to you or am I missing something?

**Comment by shd101wyy** [comment]
@kachkaev Merged. Thank you! 

Just learnt that we can start code fences with more then 3 <code>\`</code>. I used to think it requires exactly 3 <code>\`</code>.

Thank you!

**Pull request merged**